application/helpers/common_helper.php

```diff
@@ -3874,6 +3876,34 @@ function translateInsertansTags($newsid, $oldsid, $fieldnames)
     }
 }
 
+function addRegexpCondition(array $fields, $pattern = 'Q[0-9]+')
+{
+    $conditions = [];
+
+    switch (Yii::app()->db->getDriverName()) {
+        case 'mysqli':
+        case 'mysql':
+            foreach ($fields as $field) {
+                $conditions[] = "$field REGEXP '$pattern'";
+            }
+            break;
+        case 'pgsql':
+            foreach ($fields as $field) {
+                $conditions[] = "REGEXP_COUNT($field, '$pattern') > 0";
+            }
+            break;
+        case 'mssql':
+        case 'sqlsrv':
+            $pattern = str_replace('+', '%', $pattern);
+            foreach ($fields as $field) {
+                $conditions[] = "$field LIKE '$pattern'";
+            }
+            break;
+    }
+
+    return '(' . implode(' OR ', $conditions) . ')';
```

Comment:
We need to properly handle the default, that is, if the driver is something else, SQLite or something, then we need something there too.